### PR TITLE
fix: give proper error message when err.stack is undefined

### LINKED ISSUE #1669: TypeError: Cannot read property 'replace' of undefined" at Object.appendErrMsg
### Current behavior: 
This is a weird issue that is isolated to a single machine.

If I try run a test file with more than one or two tests, I get the following: 
screen shot 2018-05-03 at 10 59 12 am

The stack trace:

```
TypeError: Cannot read property 'replace' of undefined

Because this error occurred during a 'before each' hook we are skipping the remaining tests in the current suite: 'bulk actions'
    at Object.appendErrMsg (http://localhost:3000/__cypress/runner/cypress_runner.js:66723:23)
    at Runner.<anonymous> (http://localhost:3000/__cypress/runner/cypress_runner.js:65768:20)
    at Runner.EventEmitter.emit (http://localhost:3000/__cypress/runner/cypress_runner.js:28665:13)
    at Runner.fail (http://localhost:3000/__cypress/runner/cypress_runner.js:65066:17)
    at Runner.failHook (http://localhost:3000/__cypress/runner/cypress_runner.js:33083:8)
    at http://localhost:3000/__cypress/runner/cypress_runner.js:33128:16
    at http://localhost:3000/__cypress/runner/cypress_runner.js:65918:11
From previous event:
    at onNext (http://localhost:3000/__cypress/runner/cypress_runner.js:65917:57)
    at done (http://localhost:3000/__cypress/runner/cypress_runner.js:32741:5)
    at Hook.Runnable.run (http://localhost:3000/__cypress/runner/cypress_runner.js:32776:5)
    at http://localhost:3000/__cypress/runner/cypress_runner.js:65936:28
From previous event:
    at Object.onRunnableRun (http://localhost:3000/__cypress/runner/cypress_runner.js:65935:20)
    at $Cypress.action (http://localhost:3000/__cypress/runner/cypress_runner.js:61399:51)
    at Hook.Runnable.run (http://localhost:3000/__cypress/runner/cypress_runner.js:65075:20)
    at next (http://localhost:3000/__cypress/runner/cypress_runner.js:33119:10)
    at http://localhost:3000/__cypress/runner/cypress_runner.js:33141:5
    at timeslice (http://localhost:3000/__cypress/runner/cypress_runner.js:28382:27)
```

The test spec file like:

``` js
describe('tests', () => {
  beforeEach(() => {
    cy.authenticatedVisit(
      'some_url'
    )
  })
  it('draws a box', () => {})
  it('potato', () => {})
  it('draws a circle', () => {})
  it('draws nothing', () => {})
})

```
and the `authenticatedVisit` command is:

```js
Cypress.Commands.add('authenticatedVisit', url => {
  cy.visit(url, {
    onBeforeLoad() {
      window.localStorage.setItem(
        'authRefreshToken',
        userJames.authRefreshToken
      )
      window.localStorage.setItem('authAccessToken', null) //it needs this field to have an entry
    },
  })
})
```

It works perfectly fine on other machines with the exact same set up.

It works fine running a single test at a time - but I'm wondering if anyone else has experienced this before?

Happy to provide any extra info if I can!

### Versions  

Both cypress 2.1.0 and the previous version. Mac OS 10.13.4, chrome 66.0.3359.139

### PR DESCRIPTION
**Pull request by kuceb**
- Closes #1669 

### User facing changelog
- Certain failures related to Cypress commands no longer throw incorrect error 'cannot read property 'replace' of undefined'

### How has the user experience changed?
When running the following spec in **chrome**, you used to get incorrect error message, now you get correct one:
```js
describe('issue-1669 undefined err.stack in beforeEach hook', () => {

  beforeEach(() => {
    cy.setCookie('foo', '   bar')
  })

  it('cy.setCookie should fail with correct error', () => {
    expect(true).ok
  })
})

```
image

### PR Tasks

- [x] Have tests been added/updated?
- [NA] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [NA] Have API changes been updated in the [`type definitions`](cli/types/index.d.ts)?
- [NA] Have new configuration options been added to the [`cypress.schema.json`](cli/schema/cypress.schema.json)?

### CODE REVIEW

**Comment by brian-mann** [comment]
I will review this

File: packages/server/test/support/fixtures/projects/e2e/cypress/integration/issue_1669_spec.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+describe('issue-1669 undefined err.stack in beforeEach hook', () => {
+  beforeEach(() => {
+    cy.setCookie('foo', '   bar')
```
Comment (flotwig) [inline packages/server/test/support/fixtures/projects/e2e/cypress/integration/issue_1669_spec.js]: @Bkucera if you merge in `develop`, I believe this was fixed to err correctly and no longer causes an `err.stack === undefined`, you may need to come up with a custom command

Also, just curious, is there a reason why is this an `e2e` test and not just a normal driver test that asserts on the error using `cy.on('fail', (err) => {...})`?

Comment (Bkucera) [reply packages/server/test/support/fixtures/projects/e2e/cypress/integration/issue_1669_spec.js]: if you catch the error in the 'fail' handler, it has not yet hit the logic that reads off .stack 

I'll merge in develop, and try locally without these changes

Comment (Bkucera) [reply packages/server/test/support/fixtures/projects/e2e/cypress/integration/issue_1669_spec.js]: Alright, it did pass without the src changes, so I changed the test that now fails without these changes

**Review by flotwig — APPROVED** [review]

**Pull request merged**